homeassistant/components/volvooncall/__init__.py

```diff
@@ -184,32 +185,17 @@ def discover_vehicle(self, vehicle):
         self.vehicles.add(vehicle.vin)
 
         dashboard = vehicle.dashboard(
-            mutable=self.config[DOMAIN][CONF_MUTABLE],
-            scandinavian_miles=self.config[DOMAIN][CONF_SCANDINAVIAN_MILES],
+            mutable=self.config_entry.data[CONF_MUTABLE],
+            scandinavian_miles=self.config_entry.data[CONF_SCANDINAVIAN_MILES],
         )
 
         for instrument in (
             instrument
             for instrument in dashboard.instruments
             if instrument.component in PLATFORMS
         ):
-
             self.instruments.add(instrument)
-
-            self.hass.async_create_task(
-                discovery.async_load_platform(
-                    self.hass,
-                    PLATFORMS[instrument.component],
-                    DOMAIN,
-                    (
-                        vehicle.vin,
-                        instrument.component,
-                        instrument.attr,
-                        instrument.slug_attr,
-                    ),
-                    self.config,
-                )
-            )
+            dispatcher_send(self.hass, VOLVO_DISCOVERY_NEW, [instrument])
```

Comment:
We're inside the event loop.
```suggestion
            async_dispatcher_send(self.hass, VOLVO_DISCOVERY_NEW, [instrument])
```